### ISSUE TITLE
allow passing the kmod name as an `open_kmod()` argument

### ISSUE THREAD

**Issue by leogr** [falcosecurity/libs#617]
**Motivation**

The name must be hard-coded in the kmod, but not necessarily in scap.

Allowing to pass it at runtime will improve the symmetry with the ebpf (which uses the file path) and add the ability to use kmod built with other names (by other consumers).

**Feature**

- add `name` as an argument of `open_kmod()`

**Alternatives**

**Additional context**

**Comment by Andreagit97** [comment]
Yeah exactly as we do for the BPF probe, completely agree on that :+1:

**Comment by leogr** [comment]
I guess we are too late for the current milestone, so

/milestone 0.10.0

**Comment by leogr** [comment]
Not sure we still want this. Let's give it one last chance:
/remove-lifecycle stale
